Added checks for clang::Cursor::template_arg_value

cynicaldevil opened the pull request:
Fixes #134.
r? @fitzgen

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @fitzgen (or someone else) soon.

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

emilio commented:
Is there any caller in this function? If not, let's rip it out.

fitzgen commented on src/clang.rs:
```diff
@@ -457,8 +457,26 @@ impl Cursor {
+            None => false
+        };
+        match within_bounds {
+            true => {
```
In general, using `if` is usually better than `match` for boolean values:

```rust
if within_bounds {
    ...
} else {
    ...
}
```

cynicaldevil commented:
@fitzgen I've removed the method and updated the PR. Think I'll pick up another issue to work on as well, since this one turned out to be really trivial.

bors-servo commented:
:sunny: Test successful - [status-travis](https://travis-ci.org/servo/rust-bindgen/builds/173631241)

cynicaldevil commented:
@Wafflespeanut I had forgotten to to run the build step after removing the method. All good now!

Wafflespeanut commented:
@bors-servo r=emilio

bors-servo commented:
:pushpin: Commit 0e63240 has been approved by `emilio`

<!-- @bors-servo r=emilio 0e6324069148845a02dee485b49fa1c86fcd307d -->

bors-servo commented:
:hourglass: Testing commit 0e63240 with merge 09e6fe4...
